src/compositor/wayland_wrapper/qwlkeyboard.cpp

```diff
@@ -327,23 +324,38 @@ void Keyboard::createXKBKeymap()
                                          strdup(qPrintable(m_keymap.options())) };
     struct xkb_keymap *keymap = xkb_keymap_new_from_names(m_context, &rule_names, static_cast<xkb_keymap_compile_flags>(0));
 
+    if (!keymap) {
+        qWarning("Failed to load the '%s' XKB keymap.", qPrintable(m_keymap.layout()));
+        return;
```

Comment:
These return statements leak memory: free() is not called for strdup()ed strings.